src/java/generator/MapGenerator.java

```diff
@@ -117,70 +201,87 @@ public SCMap generate(long seed) throws ExecutionException, InterruptedException
+		resourceMask.combine(land).deflate(6f).combine(plateaus).minus(rock).minus(ramps).deflate(5).trimEdge(20);
+	}
 
-		BinaryMask spawnsMask = new BinaryMask(grass.getBinaryMask(), random.nextLong());
-		BinaryMask resourceMask = new BinaryMask(grass.getBinaryMask().minus(rock.getBinaryMask()), random.nextLong());
+	private void setupWreckPipeline(){
+		t1LandWreckMask = new ConcurrentBinaryMask(64, random.nextLong(), "t1LandWreck");
+		t2LandWreckMask = new ConcurrentBinaryMask(64, random.nextLong(), "t2LandWreck");
+		t3LandWreckMask = new ConcurrentBinaryMask(64, random.nextLong(), "t3LandWreck");
+		t2NavyWreckMask = new ConcurrentBinaryMask(64, random.nextLong(), "t2NavyWreck");
+		navyFactoryWreckMask = new ConcurrentBinaryMask(64, random.nextLong(), "navyFactoryWreck");
+		allWreckMask = new ConcurrentBinaryMask(513, random.nextLong(),"allWreck");
+		ConcurrentBinaryMask landCopy = new ConcurrentBinaryMask(land, random.nextLong(), "landCopy");
+
+		t1LandWreckMask.randomize(reclaimDensity *.015f).intersect(land).deflate(1).trimEdge(20);
+		t2LandWreckMask.randomize(reclaimDensity *0.01f).intersect(land).minus(t1LandWreckMask).trimEdge(20);
+		t3LandWreckMask.randomize(reclaimDensity *0.002f).intersect(land).minus(t1LandWreckMask).minus(t2LandWreckMask).trimEdge(64);
+		t2NavyWreckMask.randomize(reclaimDensity *0.015f).intersect(landCopy.outline()).trimEdge(20);
+		navyFactoryWreckMask.randomize(reclaimDensity *0.0075f).minus(land).deflate(3).trimEdge(20);
+		allWreckMask.combine(t1LandWreckMask).combine(t2LandWreckMask).combine(t3LandWreckMask).combine(t2NavyWreckMask).inflate(2);
+	}
 
-		spawnsMask.enlarge(513).minus(ramps.getBinaryMask()).deflate(18).trimEdge(20).fillCircle(256, 256, 128, false);
-		resourceMask.enlarge(513).minus(ramps.getBinaryMask()).deflate(5).trimEdge(20);
+	private void setupPropPipeline(){
+		treeMask = new ConcurrentBinaryMask(32, random.nextLong(),"tree");
+		cliffRockMask = new ConcurrentBinaryMask(32, random.nextLong(), "cliffRock");
+		fieldStoneMask = new ConcurrentBinaryMask(128, random.nextLong(), "fieldStone");
+		rockFieldMask = new ConcurrentBinaryMask(128, random.nextLong(), "rockField");
 
-		markerGenerator.generateSpawns(spawnsMask, 64);
-		markerGenerator.generateMexes(resourceMask);
-		markerGenerator.generateHydros(resourceMask);
+		cliffRockMask.randomize(.15f).intersect(rock).minus(plateaus).minus(mountains).intersect(land);
+		fieldStoneMask.randomize(reclaimDensity *.005f).enlarge(256).intersect(grass);
+		fieldStoneMask.enlarge(513).trimEdge(10);
+		treeMask.randomize(0.1f).inflate(1).cutCorners().acid(0.5f).enlarge(128).smooth(4).acid(0.5f);
+		treeMask.enlarge(256).intersect(grass);
+		treeMask.enlarge(513).deflate(5).trimEdge(3).fillCircle(256,256,96,false);
+		rockFieldMask.randomize(reclaimDensity *.005f).trimEdge(48).inflate(3).acid(.5f).intersect(land).minus(mountains);
```

Comment:
spacing, is omitting the 0 in front of the decimal place a good idea?

Reply:
Done and took out the zeros as it didnt seem to hurt.